We need to follow links in fdevent_open_cloexec
because /dev/null is symlink on Oracle Solaris.
Component lighttpd fails to start without this patch.

Will be sent to upstream.

diff --git a/src/fdevent.c b/src/fdevent.c
index bdea359..2a950d1 100644
--- a/src/fdevent.c
+++ b/src/fdevent.c
@@ -575,7 +575,11 @@ int fdevent_open_devnull(void) {
   #if defined(_WIN32)
     return fdevent_open_cloexec("nul", 0, O_RDWR, 0);
   #else
-    return fdevent_open_cloexec("/dev/null", 0, O_RDWR, 0);
+  #if defined(__sun)
+  return fdevent_open_cloexec("/dev/null", 1, O_RDWR, 0);
+  #else
+  return fdevent_open_cloexec("/dev/null", 0, O_RDWR, 0);
+  #endif
   #endif
 }
 
